datafusion/src/physical_optimizer/aggregate_statistics.rs

```diff
@@ -276,6 +336,35 @@ mod tests {
         Ok(())
     }
 
+    /// Checks that the count optimization was applied and we still get the right result
+    async fn assert_count_with_nulls_optim_success(
+        plan: HashAggregateExec,
+    ) -> Result<()> {
+        let conf = ExecutionConfig::new();
+        let optimized = AggregateStatistics::new().optimize(Arc::new(plan), &conf)?;
+
+        assert!(optimized.as_any().is::<ProjectionExec>());
```

Comment:
Maybe add a comment here that the added `ProjectionExec` is a sign the optimization was applied.